### PR TITLE
Cleanup OwnedSections component

### PR DESCRIPTION
**Pull request by TurnerRiley**
As part of moving the Owned Professional Learning Sections table from the Teacher Homepage to the My PL page in [this PR](https://github.com/code-dot-org/code-dot-org/pull/57947), `OwnedSectionsTable` had to be split into `OwnedSectionsTable` and `OwnedPlSectionsTable` so that styling could be consistent for each on their respective pages. This then left some cleanup work:
- Remove the `isPlSections` parameter to the tables
- Add a test file for `OwnedPlSectionsTable`
- While I was in here, I converted both `OwnedSectionsTableTest.js` and `OwnedPlSectionsTableTest.js` into React testing library
   - Both files run nearly identical tests, I just kept the test section data in its original file and moved the PL section data over to the pl table test file

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-1794)
PR this follows up: [here](https://github.com/code-dot-org/code-dot-org/pull/57947)

## Testing story
Updating unit tests

### CODE REVIEW

File: apps/test/unit/templates/teacherDashboard/OwnedSectionsTableTest.js [hunk 1]
```diff
@@ -260,231 +108,173 @@ describe('OwnedSectionsTable Sorting', () => {
...
-      const text = courseLinkCol.find(Button).prop('text');
-      assert.equal(button, '<Button />');
-      assert.equal(link, '/catalog');
-      assert.equal(text, 'Find a course');
-      expect(
-        courseLinkCol.find({
-          className: skeletonizeContent.skeletonizeContent,
-        })
-      ).to.have.lengthOf(0);
+  it('courseLinkFormatter provides links to course information and section information', () => {
```
Comment (hannahbergam) [inline apps/test/unit/templates/teacherDashboard/OwnedSectionsTableTest.js:209]: This test covers a lot! Non-blocking, but it feels like the format of looping through all sections and then checking that 'at least one of them' covers each of a few different states is slightly confusing and not quite future proof. I understand why you combined the concept of checking for assignmentPath counts together, but then it makes it hard to tell which section should have which count. Would it be possible to break this up?

File: apps/test/unit/templates/teacherDashboard/OwnedSectionsTableTest.js [hunk 2]
```diff
@@ -260,231 +108,173 @@ describe('OwnedSectionsTable Sorting', () => {
...
-    const wrapper = shallow(
-      <OwnedSectionsTable
-        sectionIds={[1]}
-        sectionRows={sectionRowData.slice(0, 1)}
-        onEdit={() => {}}
-      />
-    );
-    const header = wrapper.find(Table.Header);
-    assert.equal(header.length, 1);
+  it('can be sorted correctly by section name', () => {
```
Comment (hannahbergam) [inline apps/test/unit/templates/teacherDashboard/OwnedSectionsTableTest.js:137]: Love these simplified sort tests!

File: apps/test/unit/templates/teacherDashboard/OwnedPlSectionsTableTest.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+import {assert, expect} from '../../../util/reconfiguredChai';
+
+const plSectionRowData = [
+  {
+    id: 1,
+    name: 'first section',
+    studentCount: 3,
+    code: 'ABC',
+    courseOfferingsAreLoaded: true,
+    grades: ['K'],
```
Comment (hannahbergam) [inline apps/test/unit/templates/teacherDashboard/OwnedPlSectionsTableTest.js]: This is a very small thing, but do we have grades for PL sections?

**Review by hannahbergam — COMMENTED** [review]
Wow this was a ton of work! A couple questions/nits but great job!

File: apps/test/unit/templates/teacherDashboard/OwnedPlSectionsTableTest.js (hunk 3 above)
Comment (bethanyaconnor) [reply apps/test/unit/templates/teacherDashboard/OwnedPlSectionsTableTest.js]: no, actually, PL sections can [only have the grade](https://github.com/code-dot-org/code-dot-org/blob/0659952954efa4569b999231004d9117cb444455/dashboard/app/models/sections/section.rb#L121) `pl`

File: apps/test/unit/templates/teacherDashboard/OwnedPlSectionsTableTest.js [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    loginType: SectionLoginType.picture,
```
Comment (bethanyaconnor) [inline apps/test/unit/templates/teacherDashboard/OwnedPlSectionsTableTest.js]: Relatedly, they can only be email logins as well!

**Review by bethanyaconnor — APPROVED** [review]
Thanks for following up and cleaning this up!!

File: apps/test/unit/templates/teacherDashboard/OwnedSectionsTableTest.js (hunk 1 above)
Comment (TurnerRiley) [reply apps/test/unit/templates/teacherDashboard/OwnedSectionsTableTest.js:209]: Yes for sure! I'll simplify the loop tests so they're more readable and updatable in the future

File: apps/test/unit/templates/teacherDashboard/OwnedPlSectionsTableTest.js (hunk 3 above)
Comment (TurnerRiley) [reply apps/test/unit/templates/teacherDashboard/OwnedPlSectionsTableTest.js]: Oh good point! I had just copied these over from the test file and didn't notice, I'll clean that up!

File: apps/test/unit/templates/teacherDashboard/OwnedPlSectionsTableTest.js (hunk 4 above)
Comment (TurnerRiley) [reply apps/test/unit/templates/teacherDashboard/OwnedPlSectionsTableTest.js]: Ah gotcha, I'll fix that! Thanks for spotting that!

**Pull request merged**